sunpy/map/sources/soho.py

```diff
@@ -143,6 +143,13 @@ def date(self):
         time = self.meta.get('time-obs', self.meta.get('time_obs'))
         return parse_time(f"{date}T{time}")
 
+    def _set_date(self, date):
+        if 'time-obs' in self.meta:
```

Comment:
It's inconsistent that we "fix" the header here by putting the whole date&time into `date-obs` but we don't do that for GONG.

I am not sure what one we should prefer. Maybe @ayshih has thoughts.

Reply:
I wrote this, and the reason that I did it this way is because the overridden `date` property for `LASCOMap` (unlike `GONGSynopticMap`) has a logic branch anticipates that some headers will have a whole DATE-OBS.

I suppose a better approach would be to follow the original header: if it had a whole DATE-OBS, write a whole DATE-OBS; if it had a split DATE-OBS/TIME-OBS, write a split DATE-OBS/TIME-OBS.

Reply:
I pushed a rough first go at this for Gong. See what you both think.

Reply:
I think what you did for `GONGSynopticMap` looks good.  Can you add that here as well? (a few lines down)